src/main/java/dook/task/Event.java

```diff
@@ -0,0 +1,78 @@
+package dook.task;
+
+
+import java.time.LocalDate;
+import java.time.format.DateTimeParseException;
+
+import dook.services.TimeProcessor;
+
+/**
+ * Task with a start and end date.
+ */
+public class Event extends Task implements TimedTask {
+
+    protected String from;
+    protected String to;
+    protected LocalDate fromDateTime = null;
+    protected LocalDate toDateTime = null;
+    public Event(String description, String from, String to) {
+        super(description);
+        this.from = from;
+        this.to = to;
+        processDateTimes();
+    }
+
+    public Event(String description, String from, String to, boolean isDone) {
+        super(description, isDone);
+        this.from = from;
+        this.to = to;
+        processDateTimes();
+    }
+
+    @Override
+    public void processDateTimes() {
+        try {
+            fromDateTime = TimeProcessor.getLocalDateFromString(this.from);
+            toDateTime = TimeProcessor.getLocalDateFromString(this.to);
+            this.from = TimeProcessor.getStringFromLocalDate(fromDateTime);
+            this.to = TimeProcessor.getStringFromLocalDate(toDateTime);
+        } catch (DateTimeParseException e) {
+            System.out.println("Failed to parse date. Date related operations will not work on this task.");
+        }
+    }
+    @Override
+    public boolean isDuring(LocalDate dateTime) {
```

Comment:
good use of prefixes like is, has or was to denote booleans, keep it up :)

Reply:
Thanks for all of the feedback!